src/test/munit/file-read-test-case.xml

```diff
@@ -0,0 +1,174 @@
+                </file:lock-strategy>
+            </file:read>
+        </munit:execution>
+        <munit:validation>
+            <munit-tools:assert-equals actual="#[payload]" expected="Hello World!"/>
+        </munit:validation>
+    </munit:test>
+
+    <munit:test name="readWithLockTimeoutAndUnit">
+        <munit:execution>
+            <file:read config-ref="read-tests" path="text.txt">
+                <non-repeatable-stream/>
+                <file:lock-strategy>
+                    <file:with-lock lockTimeout="5000" lockTimeoutUnit="NANOSECONDS"/>
+                </file:lock-strategy>
+            </file:read>
+        </munit:execution>
+        <munit:validation>
+            <munit-tools:assert-equals actual="#[payload]" expected="Hello World!"/>
+        </munit:validation>
+    </munit:test>
+
+    <munit:test name="readWithLockTimeoutOnLockedFile">
+        <munit:execution>
+            <scatter-gather>
+                <route>
+                    <flow-ref name="writeWithLockStrategy"/>
+                </route>
+                <route>
+                    <java:invoke-static class="org.mule.extension.file.integration.FileWriteTestCase" method="sleepThread(int)">
+                        <java:args>
+                            #[{arg0: 500}]
+                        </java:args>
+                    </java:invoke-static>
+                    <file:read config-ref="read-tests" path="text.txt">
+                        <non-repeatable-stream/>
+                        <file:lock-strategy>
+                            <file:with-lock lockTimeout="30" lockTimeoutUnit="SECONDS"/>
+                        </file:lock-strategy>
+                    </file:read>
+                </route>
+            </scatter-gather>
+        </munit:execution>
+        <munit:validation>
+            <file:read config-ref="write-tests" path="text.txt"/>
+            <munit-tools:assert-equals actual="#[payload]" expected="Hello World!Hello World!"/>
+        </munit:validation>
+    </munit:test>
+
+    <munit:test name="readWithShortLockTimeoutOnLockedFileFails">
+        <munit:execution>
+            <try>
+                <scatter-gather>
+                    <route>
+                        <flow-ref name="writeWithLockStrategy"/>
+                    </route>
+                    <route>
+                        <java:invoke-static class="org.mule.extension.file.integration.FileWriteTestCase"
+                                            method="sleepThread(int)">
+                            <java:args>
+                                #[{arg0: 500}]
+                            </java:args>
+                        </java:invoke-static>
+                        <file:read config-ref="read-tests" path="text.txt">
+                            <non-repeatable-stream/>
+                            <file:lock-strategy>
+                                <file:with-lock lockTimeout="10" lockTimeoutUnit="NANOSECONDS"/>
```

Comment:
Maybe do it in a way that it tried to lock a few times here, like waiting 1 second.

Reply:
Fixed.